File: frontend/src/api/migration-audit.ts
import { client, type ApiResponse } from './client'

export interface MigrationAuditModule {
  label: string
  source_table: string
  target_table: string
  source_count: number
  target_count: number
  difference: number
  status: string
  note?: string | null
}

export interface MigrationAuditUser {
  id: number
  username: string
  display_name: string
  role: string
  enabled: boolean
  permission_snapshot_imported: boolean
}

export interface MigrationAuditSummary {
  legacy_database_configured: boolean
  legacy_database_reachable: boolean
  generated_at: string
  totals: {
    source_rows: number
    target_rows: number
    matched_modules: number
    total_modules: number
    permission_snapshots: number
  }
  modules: MigrationAuditModule[]
  users: MigrationAuditUser[]
  remarks: string[]
}

export async function fetchMigrationAuditSummary() {
  const response = await client.get<ApiResponse<MigrationAuditSummary>>('/migration-audit/summary')
  return response.data.data
}

File: backend/src/dto/migration_audit.rs
use chrono::{DateTime, Utc};
use serde::Serialize;

#[derive(Debug, Serialize)]
pub struct MigrationAuditSummaryDto {
    pub legacy_database_configured: bool,
    pub legacy_database_reachable: bool,
    pub generated_at: DateTime<Utc>,
    pub totals: MigrationAuditTotalsDto,
    pub modules: Vec<MigrationAuditModuleDto>,
    pub users: Vec<MigrationAuditUserDto>,
    pub remarks: Vec<String>,
}

#[derive(Debug, Serialize)]
pub struct MigrationAuditTotalsDto {
    pub source_rows: u64,
    pub target_rows: u64,
    pub matched_modules: u64,
    pub total_modules: u64,
    pub permission_snapshots: u64,
}

#[derive(Debug, Serialize)]
pub struct MigrationAuditModuleDto {
    pub label: String,
    pub source_table: String,
    pub target_table: String,
    pub source_count: u64,
    pub target_count: u64,
    pub difference: i64,
    pub status: String,
    pub note: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct MigrationAuditUserDto {
    pub id: u64,
    pub username: String,
    pub display_name: String,
    pub role: String,
    pub enabled: bool,
    pub permission_snapshot_imported: bool,
}
